src/eureka/S3_data_reduction/plots_s3.py

```diff
@@ -1029,17 +1070,51 @@ def phot_2d_frame(data, meta, m, i):
         plt.ylabel('y pixels')
         plt.xlabel('x pixels')
 
-        circle1 = plt.Circle((centroid_x, centroid_y), meta.photap, color='r',
-                             fill=False, lw=3, alpha=0.7,
+        # Plot proper aperture shapes
+        if meta.aperture_shape == "hexagon":
+            # to make a hexagon, make the vertices and add them into a path
+            xvert = centroid_x.data.tolist() - meta.photap*np.sin(
+                2*np.pi*np.arange(7)/6)
+            yvert = centroid_y.data.tolist() + meta.photap*np.cos(
+                2*np.pi*np.arange(7)/6)
+            hex1 = Path(np.vstack((xvert, yvert)).T)
+
+            # make patch of hexagon
+            ap1 = patches.PathPatch(hex1, color='r',
+                                    fill=False, lw=3, alpha=0.7, 
+                                    label='target aperture')
+            
+            # to make a hexagon, make the vertices and add them into a path
+            xvert = centroid_x.data.tolist() - meta.skyin*np.sin(
+                2*np.pi*np.arange(7)/6)
+            yvert = centroid_y.data.tolist() + meta.skyin*np.cos(
+                2*np.pi*np.arange(7)/6)
+            hex2 = Path(np.vstack((xvert, yvert)).T)
+
+            # make patch of hexagon
+            ap2 = patches.PathPatch(hex2, color='w',
+                                    fill=False, lw=4, alpha=0.8, 
+                                    label='sky aperture')
+            
+            # to make a hexagon, make the vertices and add them into a path
+            xvert = centroid_x.data.tolist() - meta.skyout*np.sin(
+                2*np.pi*np.arange(7)/6)
+            yvert = centroid_y.data.tolist() + meta.skyout*np.cos(
+                2*np.pi*np.arange(7)/6)
+            hex3 = Path(np.vstack((xvert, yvert)).T)
+
+            # make patch of hexagon
+            ap3 = patches.PathPatch(hex3, color='w',
+                                    fill=False, lw=4, alpha=0.8)
+        else:
+            # circular apertures
+            ap1 = plt.Circle((centroid_x, centroid_y), meta.photap, color='r',
+                             fill=False, lw=3, alpha=0.7, 
                              label='target aperture')
-        circle2 = plt.Circle((centroid_x, centroid_y), meta.skyin, color='w',
-                             fill=False, lw=4, alpha=0.8,
-                             label='sky aperture')
-        circle3 = plt.Circle((centroid_x, centroid_y), meta.skyout, color='w',
+            ap2 = plt.Circle((centroid_x, centroid_y), meta.skyin, color='w',
+                             fill=False, lw=4, alpha=0.8, label='sky aperture')
+            ap3 = plt.Circle((centroid_x, centroid_y), meta.skyout, color='w',
                              fill=False, lw=4, alpha=0.8)
-        plt.gca().add_patch(circle1)
-        plt.gca().add_patch(circle2)
-        plt.gca().add_patch(circle3)
```

Comment:
Why remove these three lines, rather than replace them like you did with the `ap1`, `ap2`, and `ap3` for Fig 3306 above?

Reply:
Sorry, that was a mistake. They're replaced now.